presto-main/src/main/java/com/facebook/presto/version/EmbedVersion.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.version;
+
+import com.facebook.presto.server.ServerConfig;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.bytecode.ClassDefinition;
+import io.airlift.bytecode.FieldDefinition;
+import io.airlift.bytecode.MethodDefinition;
+import io.airlift.bytecode.Parameter;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Objects;
+
+import static com.facebook.presto.util.CompilerUtils.defineClass;
+import static com.facebook.presto.util.CompilerUtils.makeClassName;
+import static com.facebook.presto.util.Reflection.constructorMethodHandle;
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static io.airlift.bytecode.Access.FINAL;
+import static io.airlift.bytecode.Access.PRIVATE;
+import static io.airlift.bytecode.Access.PUBLIC;
+import static io.airlift.bytecode.Access.a;
+import static io.airlift.bytecode.Parameter.arg;
+import static io.airlift.bytecode.ParameterizedType.type;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class EmbedVersion
+{
+    private final String builtInVersion;
+    private final String configuredVersion;
+
+    private MethodHandle runnableConstructor;
+
+    @Inject
+    public EmbedVersion(ServerConfig serverConfig)
+    {
+        this.builtInVersion = new ServerConfig().getPrestoVersion();
+        this.configuredVersion = serverConfig.getPrestoVersion();
+    }
+
+    @PostConstruct
```

Comment:
why not in constructor?

Reply:
I think this is better place guice-wise. i don't have strong opinion though.